flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/IOMetrics.java

```diff
@@ -43,38 +43,74 @@
 	protected double numBytesInRemotePerSecond;
 	protected double numBytesOutPerSecond;
 
-	public IOMetrics(Meter recordsIn, Meter recordsOut, Meter bytesLocalIn, Meter bytesRemoteIn, Meter bytesOut) {
+	protected long numBuffersInLocal;
+	protected long numBuffersInRemote;
+	protected long numBuffersOut;
+
+	protected double numBuffersInLocalPerSecond;
+	protected double numBuffersInRemotePerSecond;
+	protected double numBuffersOutPerSecond;
+
+	public IOMetrics(
```

Comment:
so this here is really the big question mark for me. So far we only kept metrics in here that are displayed in the UI, but with this PR we'd break that. This is mostly done to have accurate displays at the end of the job, which in particular in the case of batch jobs is important.

The question is, do we truly need 100% accurate buffer metrics here? For _debugging_ purposes there are always other avenues one can take to get accurate readings. (delay shutdown of job until metrics are updated)

Reply:
ok - good to know.

Since there was no description of what needs to be where, I simply added the buffer metrics anywhere I saw the byte and record metrics.
-> So I can/should remove them from `IOMetrics` and related classes and that should be fine?

Reply:
it should be fine to remove them from this class, and revert the changes to the various handlers.

Reply:
that's what I meant - done